Migrate `sci` plugin to `builtins` (#704)
Green-Software-Foundation/if: src/builtins/sci/README.md

  the number of requests per `functional-unit`.
- `functional-unit-time`: a time value and a unit as a single string.
  E.g. `2 s`, `10 seconds`, `3 days`, `2 months`, `0.5 y`.

In a plugin pipeline, time is always denominated in `seconds`. It is only in
`sci` that other units of time are considered. Therefore, if `functional-unit-time`
is `1 month`, then the sum of `carbon-operational` and `carbon-embodied` is
multiplied by the number of seconds in one month.

Example:

```yaml
carbon-operational: 0.02  // carbon-operational per s
carbon-embodied: 5   // carbon-embodied per s
functional-unit: requests  // indicate the functional unit is requests
functional-unit-time: 1 minute  // time unit is minutes
requests: 100   // requests per minute
```

```pseduocode
sci-per-s = carbon-operational + carbon-embodied / duration  // (= 5.02)
sci-per-minute = sci-per-s * 60  // (= 301.2)
sci-per-functional-unit-time = sci-per-minute * number of minutes
sci-per-f-unit = sci-per-functional-unit-time / 100  // (= 3.012 gC/request)
```

To run the plugin, you must first create an instance of `Sci`. Then, you can call `execute()` to return `sci`.

```typescript

> import {Sci} from '@grnsft/if-builtin';

Author: ```suggestion
import {Sci} from 'builtins';
```

const sci = Sci({'functional-unit': 'requests'});
const results = await sci.execute(
  [
    {
      'carbon-operational': 0.02,
      'carbon-embodied': 5,
      duration: 1,
      requests: 100,
    },
  ],
  {
    'functional-unit-time': '1 day',
  }
);
```

## Example manifest

IF users will typically call the plugin as part of a pipeline defined in a `manifest`
file. In this case, instantiating the plugin is handled by
`ie` and does not have to be done explicitly by the user.
The following is an example `manifest` that calls `sci`:

```yaml
name: sci-demo
description: example invoking sci plugin
tags:
initialize:
  outputs:
    - yaml
  plugins:
    sci:
      method: Sci

>       path: 'builtins'

Reviewer: should be `builtin`

Author: ```suggestion
      path: 'builtin'
```

>       path: '@grnsft/builtin'

Author: ```suggestion
      path: 'builtin'
```

  the number of requests per `functional-unit`.
- `functional-unit-time`: a time value and a unit as a single string.
  E.g. `2 s`, `10 seconds`, `3 days`, `2 months`, `0.5 y`.

In a plugin pipeline, time is always denominated in `seconds`. It is only in
`sci` that other units of time are considered. Therefore, if `functional-unit-time`
is `1 month`, then the sum of `carbon-operational` and `carbon-embodied` is
multiplied by the number of seconds in one month.

Example:

```yaml
carbon-operational: 0.02  // carbon-operational per s
carbon-embodied: 5   // carbon-embodied per s
functional-unit: requests  // indicate the functional unit is requests
functional-unit-time: 1 minute  // time unit is minutes
requests: 100   // requests per minute
```

```pseduocode
sci-per-s = carbon-operational + carbon-embodied / duration  // (= 5.02)
sci-per-minute = sci-per-s * 60  // (= 301.2)
sci-per-functional-unit-time = sci-per-minute * number of minutes
sci-per-f-unit = sci-per-functional-unit-time / 100  // (= 3.012 gC/request)
```

To run the plugin, you must first create an instance of `Sci`. Then, you can call `execute()` to return `sci`.

```typescript

> import {Sci} from '@grnsft/if-plugins';

Reviewer: change import

Author: ```suggestion
import {Sci} from '@grnsft/if-builtin';
```

const sci = Sci({'functional-unit': 'requests'});
const results = await sci.execute(
  [
    {
      'carbon-operational': 0.02,
      'carbon-embodied': 5,
      duration: 1,
      requests: 100,
    },
  ],
  {
    'functional-unit-time': '1 day',
  }
);
```

## Example manifest

IF users will typically call the plugin as part of a pipeline defined in a `manifest`
file. In this case, instantiating the plugin is handled by
`ie` and does not have to be done explicitly by the user.
The following is an example `manifest` that calls `sci`:

```yaml
name: sci-demo
description: example invoking sci plugin
tags:
initialize:
  outputs:
    - yaml
  plugins:
    sci:
      method: Sci

>       path: '@grnsft/if-plugins'

Reviewer: can you change the path here too?

Author: ```suggestion
      path: '@grnsft/builtin'
```

      global-config:
        functional-unit-time: '5 minutes'
tree:
  children:
    child:
      pipeline:
        - sci
      config:
        sci:
          functional-unit: requests # factor to convert per time to per f.unit
      inputs:
        - timestamp: 2023-07-06T00:00
          carbon-operational: 0.02
          carbon-embodied: 5
          duration: 1
          requests: 100
```

> You can run this example `manifest` by saving it as `./examples/manifests/test/sci.yml` and executing the following command from the project root:

Reviewer: change manifest path

Author: ```suggestion
You can run this example `manifest` by saving it as `./manifests/plugins/sci.yml` and executing the following command from the project root:
```

```sh
npm i -g @grnsft/if

> npm i -g @grnsft/if-plugins

Reviewer: remove this line
